internal/modules/modules.go

```diff
@@ -96,16 +98,214 @@ type ModuleResolveOptions struct {
 //
 //nolint:funlen // Why: Will be refactored in the future
 func GetModulesForService(ctx context.Context, opts *ModuleResolveOptions) ([]*Module, error) {
-	// start resolving the top-level modules
-	modulesToResolve := make([]resolveModule, 0)
+	wl := buildWorkLists(opts)
+
+	log := opts.Log
+
+	concurentResolutions := 5
```

Comment:
Should probably be flag driven with a default

Reply:
fair

Reply:
done